app/models/account.rb

```diff
@@ -540,6 +544,13 @@ def journal_entry_items_calculate(column, started_at, stopped_at, operation = :s
     journal_entry_items.where(printed_on: started_at..stopped_at).calculate(operation, column)
   end
 
+  def is_number_valid?
```

Comment:
Why not use a something like:
```
def is_number_valid?
  digits = self.number.split
  <your condition>
end
```
Instead of using unnecessary variables and a `false if` ?